pkg/plugin/datasource.go

```diff
@@ -38,25 +38,26 @@ var (
 func NewDatasource(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
 	apiKey := settings.DecryptedSecureJSONData["apiKey"]
 
-	spice := gospice.NewSpiceClient()
-
 	if apiKey == "" {
 		return nil, fmt.Errorf("missing Spice AI apiKey")
 	}
 
-	if err := spice.Init(apiKey); err != nil {
-		return nil, fmt.Errorf("failed to initialize gospice: %w", err)
+	spice := gospice.NewSpiceClient()
+
+	if err := spice.Init(
+		gospice.WithApiKey(apiKey),
+		gospice.WithSpiceCloudAddress(),
+	); err != nil {
+		return nil, fmt.Errorf("failed to initialize SpiceClient: %w", err)
 	}
 
 	opts, err := settings.HTTPClientOptions(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("http client options: %w", err)
 	}
-	opts.Headers = map[string]string{
-		"Content-Type":    "application/json",
-		"Accept-Encoding": "gzip, deflate",
-		"X-API-Key":       apiKey,
-	}
+	opts.Header.Add("Content-Type", "application/json")
+	opts.Header.Add("Accept-Encoding", "gzip, deflate")
+	opts.Header.Add("X-API-Key", apiKey)
```

Comment:
@sgrebnov let's add a client id string. @phillipleblanc is working on the telemetry spec and can provide the header.

Reply:
Update on telemetry  - @phillipleblanc provided me with the future header strings but I've realized that we don't have this logic yet so I can't configure Spice - I'll do this later once Spice supports telemetry / client configuration.